src/async.ts

```diff
@@ -270,8 +270,11 @@ interface BaseBuild<T> {
 
 async function buildBase<T, K extends keyof T>(
   seqNum: number,
-  builder: Builder<T, K>
+  builder: Builder<T, K> | Promise<Builder<T, K>>
 ): Promise<BaseBuild<T>> {
+  if (isPromise(builder))
+    builder = await builder;
```

Comment:
I generally don't like reassignment like this, although Typescript does helpfully update the known type of `builder`.  The `lift()` function is meant to help with situations like this:
```
const resolvedBuilder = await lift(builder);
// now use resolvedBuilder going forward
```